tracee-rules/benchmark/signature/rego/aio.rego

```diff
@@ -0,0 +1,260 @@
+package tracee.aio
```

Comment:
Alternative solution without changing the interface between engine and signature definition might be to create aggregating signature and leverage incremental rules and then compile it with existing signatures provided by users:

```rego
# Compile all.rego with existing Rego signatures
package tracee.all

tracee_match = res {
    # If res allowed returning multiple detections, we could
    # reuse existing rego.NewSignature constructor and pass all
    # Rego source including this one
    res := {
        "detections": detect
    }
}

# detect is an incremental rule so we can evaluate one to many
# existing signatures
detect[resp] {
	resp := {
		"name": data.tracee.TRC_2.__rego_metadoc__.name,
		"matches": data.tracee.TRC_2.tracee_match
	}
}

detect[resp] {
	resp := {
		"name": data.tracee.TRC_3.__rego_metadoc__.name,
		"matches": data.tracee.TRC_3.tracee_match
	}
}

# Define detect[res] for all signatures that you want to evaluate
```

You could instantiate it as any existing Rego signature without commenting out code in other places:

```go
regosig.NewRegoSignature(helpersRego, codeInjectionRego, antiDebuggingPtracemeRego, allRego)
```

However, this would require logic to parse result set with zero to many detections in the `OnEvent` callback.